joplin/base/wagtail_hooks.py

```diff
@@ -300,6 +301,158 @@ def register_link_handler(features):
     features.register_link_type(InternalLinkHandler)
 
 
+# In here we're going to remake all the urls for this page
+@hooks.register('after_edit_page')
+def after_edit_page(request, page):
```

Comment:
I realize this is a first attempt, but ideally what I'd like to see is the functionally broken into it's own functions (with tests), and then those functions get called on the hook here.

Reply:
I agree, and I think that fits in nicely with https://github.com/cityofaustin/techstack/issues/3630, so we can just factory up pages with relationships and then make sure we get the expected URLs